### PR TITLE
Fix for JPTFastjet correction chains

### PR DESCRIPTION
**Pull request by kodolova**
Correction of the L1Fstajet correction for JPT

### CODE REVIEW

**Comment by perrotta** [comment]
@kodolova please edit the PR title to include explicitly the description of the fix

**Comment by slava77** [comment]
@cmsbuild please test

**Comment by kodolova** [comment]
Made change in JetMETCorrections/Configuration/python/JetCorrectorsAllAlgos_cff.py
There was an artefact with ak7JPTL1OffsetCorrector. Changed to  ak7CaloL1OffsetCorrector.

**Comment by slava77** [comment]
@cmsbuild please test

**Comment by kodolova** [comment]
I have added  ak7L1JPTFastjetCorrector to JetCorrectorsAllAlgos_cff.py. Checked with runTheMatrix.py -s --useInput all > & log

**Comment by slava77** [comment]
@cmsbuild please test

**Comment by kodolova** [comment]
Correct me if I am wrong. This error has nothing to do with JetMETCorrections/Configuration. 
An exception of category 'FatalRootError' occurred while
   [0] Processing run: 1 lumi: 1 event: 1
   [1] Running path 'pathALCARECOMuAlZMuMuGeneralTracks'
   [2] Calling method for module TrackingMonitor/'ALCARECOMuAlZMuMuGeneralTracksTrackingDQM'
   Additional Info:
      [a] Fatal Root Error: @SUB=TClingCallFunc::make_wrapper
Failed to compile

**Comment by slava77** [comment]
@cmsbuild please test

error in 25.0 is apparently transient.

**Comment by perrotta** [comment]
@kodolova could you  please attach here the address of a presentation (group meeting or anywhere else) in which the effect of this "bug fix" was shown and/or discussed?
Could you please also edit the title of this PR and modify it into something more descriptive, so that it will  be more easily identified in the list of the PRs integrated in CMSSW?

**Comment by kodolova** [comment]
There is a PAS that shows how to use L1Offset/Fastjet for JPT:https://cms.cern.ch/iCMS/analysisadmin/get?analysis=JME-14-005-pas-v30.pdf (page 3). L1JPTFastjet corrections should be done in the same way as L1JPTOffset. Otherwise it does not pickup ZSP corrections for JPT.

**Comment by kodolova** [comment]
I renamed to L1JPTFastjetFix.

**Comment by slava77** [comment]
On 3/18/17 7:54 AM, kodolova wrote:
> I renamed to L1JPTFastjetFix.

A grammatically correct title in plain English would be nice.
Thank you.

File: JetMETCorrections/Configuration/python/JetCorrectors_cff.py [hunk 1]
```diff
@@ -48,7 +47,16 @@
...
+#### JPT L1Fastjet need L1CaloFastjet
+ak4L1JPTFastjetCorrector = cms.EDProducer(
+    'L1JPTOffsetCorrectorProducer',
+    level = cms.string('L1JPTOffset'),
+    algorithm = cms.string('AK4JPT'),
+    offsetService = cms.InputTag('ak4CaloL1FastjetCorrector')
+    )
+ak4L1JPTFastjetCorrectorChain = cms.Sequence(
+    ak4CaloL1FastjetCorrector * ak4L1JPTFastjetCorrector
+)
```
Comment (monttj) [inline JetMETCorrections/Configuration/python/JetCorrectors_cff.py]: @kodolova 

I am wondering if this sequence is really used somewhere?
"ak4L1JPTFastjetCorrectorChain"

Regards,
Taejeong

**Comment by kodolova** [comment]
This is the sequence for L1JPTFastjet corrections. Please, look to CorrectedJetProducers_cff.py
ak4JPTJetsL1 = cms.EDProducer(
    'CorrectedJPTJetProducer',
    src        = cms.InputTag('JetPlusTrackZSPCorJetAntiKt4'),
    correctors = cms.VInputTag('ak4L1JPTFastjetCorrector')
    )
This is the correct producer for L1JPTFastjet offset.

**Comment by perrotta** [comment]
@kodolova : this PR does not merge any more: could you please rebase and/or update it?
Furthermore: do you have any plot that shows the expected changes because of the fix implemented here? Or can you tell us what would you actually expect?

**Comment by kodolova** [comment]
The reason of change is that JPTjets have the L1 corrections in two steps: firstly Calojets (part of JPTjet) are corrected for L1Offset or L1Fastjet and then corrections for zero suppression is done. Currently it is correctily implemented only for L1Offset. For chains that includes CaloL1Fastjet, zsp corrections are missed and this is 20% of jet energy scale. 
  Concerning rebase/update, I see that Configuration is completely changed. Thus, this is not a matter of rebase. I sent message to JetMET conveners and ask how to proceed.

**Comment by kodolova** [comment]
I am closing this PR and open the new one.

**Comment by kodolova** [comment]
The new PR is 18121
